internal/setup/openpgp.go

```diff
@@ -0,0 +1,92 @@
+package setup
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"golang.org/x/crypto/openpgp/armor"
+	"golang.org/x/crypto/openpgp/clearsign"
+	"golang.org/x/crypto/openpgp/packet"
+)
+
+// DecodeKeys decodes public and private key packets from armored data.
+func DecodeKeys(armoredData []byte) (pubKeys []*packet.PublicKey, privKeys []*packet.PrivateKey, err error) {
+	block, err := armor.Decode(bytes.NewReader(armoredData))
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot decode armored data")
+	}
+
+	reader := packet.NewReader(block.Body)
+	for {
+		p, err := reader.Next()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, nil, err
+		}
+		if privKey, ok := p.(*packet.PrivateKey); ok {
+			privKeys = append(privKeys, privKey)
+		}
+		if pubKey, ok := p.(*packet.PublicKey); ok {
+			pubKeys = append(pubKeys, pubKey)
+		}
+	}
+	return pubKeys, privKeys, nil
+}
+
+// DecodeArchivePublicKey decodes a single public key packet from armored data.
+// The data should contain exactly one public key packet and no private key packets.
+func DecodeArchivePublicKey(armoredData []byte) (*packet.PublicKey, error) {
```

Comment:
Can we rename this to `DecodePublicKey`? While it is true that we are using it for Archive the function and its implementation are more general than that.